File: 11.Greedy/nonOverlap.1.py
def nonOverlapIntervals(self, A):
    if not A or len(A) == 0:
        return 0

    A.sort(key=lambda x: x[1])

    preEnd = float('-inf')
    ans = 0

    for r in A:
        start = r[0]
        end = r[1]

        if preEnd <= start:
            preEnd = end
            ans += 1

    return ans

File: 11.Greedy/nonOverlap.1.java
public int nonOverlapIntervals(int[][] A) {
  final int N = A == null ? 0 : A.length;

  Arrays.sort(A, new Comparator<int[]>() {
    public int compare(int[] a, int[] b) {
      return a[1] == b[1] ? 0 : (a[1] < b[1] ? -1 : 1);
    }
  });

  int preEnd = Integer.MIN_VALUE;
  int ans = 0;

  for (int i = 0; i < N; i++) {
    final int start = A[i][0];
    if (preEnd <= start) {
      preEnd = A[i][1];
      ans++;
    }
  }

  return ans;
}
